### DISCUSSION TITLE
Geomagic not working v22.06

### DISCUSSION THREAD

**Discussion by griffinandrew** [sofa-framework/sofa#3293] [Haptics] [answered]
Hi,

I am trying to use the geomagic plugin demo examples, yet, I am unable to use my haptic device in the scene. I downloaded and installed the new version of SOFA (v22.06) with the precompiled geomagic binary. I have loaded the geomagic plugin using the plugin manager. Yet, when I try to load the basic demo scene the haptic device does not appear as seen in the attached image. The component state is invalid as well as seen with the attached image and the output of the executable is attached. This was also an issue with the previous version, but when I built SOFA from scratch and enabled the plugin in cmake this was not an issue. Please advise as to best to proceed. Thank you! 

-Griffin

![image](https://user-images.githubusercontent.com/76177337/190024927-5209dcf7-cedd-4129-a2a4-c18d82f158dd.png)
![image](https://user-images.githubusercontent.com/76177337/190024944-4662feff-0ae6-4d1b-afdf-d2bb6a18820d.png)

`[INFO]    [runSofa] PluginRepository paths = C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin;C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/plugins;C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/collections;C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin
[INFO]    [runSofa] DataRepository paths = C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/share/sofa;C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/share/sofa/examples;C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/share/sofa;C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/share/sofa/examples
[INFO]    [runSofa] GuiDataRepository paths = C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/share/sofa/gui/runSofa;C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/share/sofa/gui/qt;C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/share/sofa/gui/qt
[INFO]    [runSofa] Loading automatically plugin list in C:\Users\jrogat\SOFA\SOFA_v22.06.00_Win64\bin/plugin_list.conf
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.ODESolver.Forward.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.ODESolver.Backward.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.ODESolver.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.IO.Mesh.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.IO.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Playback.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.SceneUtility.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Topology.Container.Constant.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Topology.Container.Dynamic.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Topology.Container.Grid.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Topology.Container.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Topology.Mapping.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Topology.Utility.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Topology.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Visual.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.LinearSolver.Iterative.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.LinearSolver.Direct.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.LinearSolver.Preconditioner.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.LinearSolver.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Mass.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Diffusion.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Mapping.Linear.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Mapping.NonLinear.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Mapping.MappedMatrix.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Mapping.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.SolidMechanics.Spring.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.SolidMechanics.FEM.Elastic.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.SolidMechanics.FEM.HyperElastic.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.SolidMechanics.FEM.NonUniform.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.SolidMechanics.FEM.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.SolidMechanics.TensorMass.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.SolidMechanics.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.StateContainer.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Constraint.Lagrangian.Model.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Constraint.Lagrangian.Correction.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Constraint.Lagrangian.Solver.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Constraint.Lagrangian.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Constraint.Projective.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Constraint.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.AnimationLoop.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.MechanicalLoad.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Collision.Geometry.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Collision.Detection.Algorithm.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Collision.Detection.Intersection.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Collision.Detection.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Collision.Response.Mapper.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Collision.Response.Contact.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Collision.Response.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Collision.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Setting.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Controller.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Engine.Analyze.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Engine.Generate.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Engine.Select.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Engine.Transform.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Engine.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.Haptics.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.Component.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.GL.Component.Rendering2D.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.GL.Component.Rendering3D.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.GL.Component.Engine.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.GL.Component.Shader.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.GL.Component.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Sofa.GUI.Component.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/SceneCreator.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/CImgPlugin.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/SofaMatrix.dll
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/BeamAdapter.dll
[INFO]    [PluginManager] 68 plugins have been loaded from C:\Users\jrogat\SOFA\SOFA_v22.06.00_Win64\bin/plugin_list.conf
[INFO]    [RealGUI] Set multisampling anti-aliasing (MSAA) with 16 samples.
[INFO]    [SofaPluginManager] Loading automatically plugin list in C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/config/loadedPlugins.ini
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/Geomagic.dll
[INFO]    [SofaPython3] Initializing with python version 3.8.10 (tags/v3.8.10:3d8993a, May  3 2021, 11:48:03) [MSC v.1928 64 bit (AMD64)]
[INFO]    [SofaPython3] Registering a scene loader for [.py, .py3, .pyscn, .py3scn] files.
[INFO]    [SofaPython3] Initializing python
[INFO]    [SofaPython3] Added 'C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/plugins/SofaPython3/lib/python3/site-packages' to sys.path
---------------------------------------
Checking SOFA_ROOT and SOFAPYTHON3_ROOT
Using environment variable SOFA_ROOT: C:\Users\jrogat\SOFA\SOFA_v22.06.00_Win64
Warning: environment variable SOFAPYTHON3_ROOT is empty. Trying to guess it.
Guessed SOFAPYTHON3_ROOT: C:\Users\jrogat\SOFA\SOFA_v22.06.00_Win64\plugins\SofaPython3
---------------------------------------
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/plugins/SofaPython3/bin/SofaPython3.dll
[INFO]    [PluginManager] 2 plugins have been loaded from C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/config/loadedPlugins.ini
[WARNING] [Qt] The constructor with a shareWidget is deprecated, use the regular contructor instead.

[INFO]    [SceneCheckerVisitor] Validating node "root" with checks: [SceneCheckDuplicatedName, SceneCheckMissingRequiredPlugin, SceneCheckUsingAlias, SceneCheckDeprecatedComponents, SceneCheckCollisionResponse]
[INFO]    [SceneCheckerVisitor] Finished validating node "root".
[DEPRECATED] [SofaOpenglVisual] SofaOpenglVisual is deprecated;. Use Sofa.GL.Component.Rendering2D, Sofa.GL.Component.Rendering3D and Sofa.GL.Component.Shader instead.
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/SofaOpenglVisual.dll
[DEPRECATED] [SofaConstraint] SofaConstraint is deprecated. It will be removed at v23.06. You may use Sofa.Component.Mapping.MappedMatrix, Sofa.Component.Constraint.Lagrangian.Model, Sofa.Component.Constraint.Lagrangian.Correction, Sofa.Component.Constraint.Lagrangian.Solver, Sofa.Component.AnimationLoop, Sofa.Component.Collision.Detection.Intersection and Sofa.Component.Collision.Response.Contact instead.
[INFO]    [PluginManager] Loaded plugin: C:/Users/jrogat/SOFA/SOFA_v22.06.00_Win64/bin/SofaConstraint.dll
[INFO]    [SceneCheckerVisitor] Validating node "root" with checks: [SceneCheckDuplicatedName, SceneCheckMissingRequiredPlugin, SceneCheckUsingAlias, SceneCheckDeprecatedComponents, SceneCheckCollisionResponse]
[INFO]    [SceneCheckerVisitor] Finished validating node "root".
[WARNING] [GeomagicDriver(GeomagicDevice)] No forceFeedBack component found in the scene. Only the motion of the haptic tool will be simulated.`

**Comment by fredroy** [comment, marked as answer]
As far as I know, the geomagic plugin in the archive from the web site is not compiled using the OpenHaptics SDK, therefore it cannot communicate with geomagic devices (because of licensing/distrubuting issues maybe?).
You will need to compile it by yourself (and you will need to download the SDK with your own means)

**Reply by griffinandrew** [reply]
Thank you!
